docs/contributions-flow.md

```diff
@@ -14,19 +14,51 @@ See the License for the specific language governing permissions and
 limitations under the License.
 -->
 
-# SwG Contributions Flow
+# SwG Contributions Flow (Experiments ONLY)
 
-This flow allows the publication site to display a flow where users can contribute money to
-the publisher. See [Subscriptions APIs](./core-apis.md).
+This flow allows the publication site to display a flow where users can contribute money to the publisher. See [Contributions APIs](./core-apis.md).
 
+First, please setup the contribution response callback via `setOnContributionResponse`:
 
-The contribution flow will first present a set of amounts user can contribute to.
-A user will get a choice to either select one of the offers, or try request login to claim an existing contribution.
+```
+subscriptions.setOnContributionResponse(function(subscriptionPromise) {
+  subscriptionPromise.then(function(response) {
+    // 1. Handle contribution response.
+    // 2. Once contribution is processed (account created):
+    response.complete().then(function() {
+      // 3. The contribution is fully processed.
+    });
+  });
+});
+```
+
+Once you receive contribution response:
+ 1. You can process the contribution. For instance you can create a new account based on the `userData` info and save contribution for this account.
+ 2. Once contribution is processed, call `response.complete()`. This method will signal to SwG that your site has accepted the contribution. It will return a promise that will be resolved once the user has acknowledged contribution.
+ 3. Once the `response.complete()` promise is resolved, you can unblock content, show additional UI to the user or perform any actions you see fit.
+
+To activate contribution flow itself, call the `contribute` method with the desired SKU:
+
+```
+subscriptions.contribute(sku);
+```
+
+The `setOnContributionResponse` callback will be called once the contribution is complete, or when the previously executed contribution is recovered.
+
+Another way to trigger the contributions flow is by first present a dialog with a set of amounts presented to the user to contribute to.
+
+A user will get a choice to either select one of the amounts to contribute to, or try request login to claim an existing contribution. This feature may not be available initially.
 
 To display contributions:
 
 ```
-subscriptions.showContributions();
+subscriptions.showContributionOptions();
+```
+
+If a user elects for a presented contribution options, publisher calls `showContributions` where The contributions API (`showContributionOptions`) accepts a list of SKUs to be displayed. The list of SKUs should be of type type `UI_CONTRIBUTIONS` (configuration).
```

Comment:
"If a user elects for a presented contribution options" <-- I also don't know what this means.

Reply:
Re-format/Re-factored the comment to make it more clear.